src/auth/scopes/index.ts

```diff
@@ -4,11 +4,11 @@ class AuthScopes {
   private compressedScopes: Set<string>;
   private expandedScopes: Set<string>;
 
-  constructor(scopes: string | string[]) {
+  constructor(scopes: string | string[] | undefined) {
     let scopesArray: string[] = [];
     if (typeof scopes === 'string') {
       scopesArray = scopes.split(new RegExp(`${AuthScopes.SCOPE_DELIMITER}\\s*`));
-    } else {
+    } else if (scopes) {
```

Comment:
Should we be extra sure that we only process accepted types?

```suggestion
    } else if (scopes !== undefined) {
```

Reply:
That's a good point. Should there be an exception/error at that point if `scopes` is not a string, array or undefined?

Reply:
Technically, typescript would fail in that case, but javascript wouldn't. Then again, we don't check for that sort of thing in a lot of other places (and is that an excuse for not doing it here? 😅 ).

Reply:
I think an easy way to handle something like this is using something like [`lodash.castArray`](https://lodash.com/docs/#castArray) so that we're always dealing with a list, so:

```jsx
// type any for illustration
constructor(scopes: any) {
  let scopesArray = castArray(scopes).filter(Boolean)
}
```